src/rp2_common/hardware_rtc/rtc.c

```diff
@@ -11,9 +11,28 @@
 #include "hardware/resets.h"
 #include "hardware/clocks.h"
 
+#define RANGE_CHECK_YEAR(t)  (t->year >= 0 && t->year <= 4095)
+#define RANGE_CHECK_MONTH(t) (t->month >= 1 && t->month <= 12)
+#define RANGE_CHECK_DAY(t)   (t->day >= 1 && t->day <= 32)
```

Comment:
Which months have 32 days? :wink:

Reply:
Oh there are months that won't end at all :)
I was not sure whether <= or < has better readability, and so this slipped through.